### DISCUSSION TITLE
Resume training from checkpoint

### DISCUSSION THREAD

**Discussion by jzallen07** [explosion/spaCy#7245] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour

Using Spacy 3. Train a model until a checkpoint is produced e.g. the model-best output. Seemingly no way to resume training from the produced checkpoint. I've tried using the `source` parameter in the config file and get an error saying that the path is not a package error. There is some documentation around resuming training in the PRE training method but nothing for the general training method and no apparent way to provide a path to the config file other than initializing a new NLP object to train from. 

## Your Environment

- **spaCy version:** 3.0.3
- **Platform:** Linux-5.3.0-1035-aws-x86_64-with-Ubuntu-18.04-bionic
- **Python version:** 3.6.9
- **Pipelines:** en_core_web_lg (3.0.0)

**Comment by adrianeboyd** [comment, marked as answer]
To resume training, `source` is the right answer. The `source` should be something that can be loaded with `spacy.load()`, either a package name like `en_core_web_sm` or a path to a model directory.

Let's say you wanted to resume training a textcat model. You can make a copy of the `config.cfg` from `model-best` and edit it to source the textcat component instead of defining it from scratch. For the textcat component section, your config would only contain the following:

```ini
[components.textcat]
source = "/path/to/model-best"
```

And then `spacy train` will show:

```none
[2021-03-02 09:10:18,717] [INFO] Set up nlp object from config
[2021-03-02 09:10:18,726] [INFO] Pipeline: ['textcat']
[2021-03-02 09:10:18,726] [INFO] Resuming training for: ['textcat']
```

If there are components you want to add from a source but not update, then add them to `[training.frozen_components]` so they're copied into the new pipeline but then not modified by the train step.
